xonsh/parsers/base.py

```diff
@@ -19,13 +22,55 @@
 from xonsh.tokenize import SearchPath, StringPrefix
 from xonsh.lazyasd import LazyObject
 from xonsh.parsers.context_check import check_contexts
+from xonsh.events import events
+from xonsh.tools import print_exception
```

Comment:
> Or, alternatively, we can say transformers only apply to interactive contexts.

This has been my thought all along. The only transformations we have been asked for don't really make sense in a script environment. This could change if we try to reimagine the context sensitive transformer as a transform event. Then it would need to be applied to scripts too.I was hoping to worry about that later though :)

I agree about the history backends too.